_include/_ncidef.h

```diff
@@ -0,0 +1,18 @@
+#ifndef INTERNAL_NCIDEF_H
+#define INTERNAL_NCIDEF_H
+/**************************************
+  _NCIDEF.H - definitions of constants
+  used in  to define compression method 
+  names and routines.
+**************************************/
+
+#define DEFINE_COMPRESSION_METHOD_STRINGS \
+    static const char *compression_methods[] = {"standard", "gzip"};
```

Comment:
I agree we should make an array of images.  Since we are checking the index against the length of the array, I see no reason to make the array bigger than it is.